File: algorithms/1822.sign-of-the-product-of-an-array/solution.go
func arraySign(nums []int) int {
	negativeCount := 0
	for _, n := range nums {
		if n == 0 {
			return 0
		}
		if 0 > n {
			negativeCount += 1
		}
	}

	if negativeCount%2 == 0 {
		return 1
	} else {
		return -1
	}
}

File: algorithms/1822.sign-of-the-product-of-an-array/solution.rb
# @param {Integer[]} nums
# @return {Integer}
def array_sign(nums)
  negative_count = 0
  nums.each do |n|
    return 0 if n == 0
    negative_count += 1 if 0 > n
  end
  negative_count % 2 == 0 ? 1 : -1
end
